plugin.video.telebasel/resources/lib/telebasel.py

```diff
@@ -269,29 +269,34 @@ def retrieve_shows(self):
 
         shows = []
         image_regex = r'data-bg\s*=\s*\"url\(\'(?P<url>.+)\''
+        filename_regex = r'(?P<trunk>.+)-\d+x\d+\.(?P<ext>.+)'
         for ssoup in shows_soup:
-            page = ssoup.attrs['href']
-            m = re.search(image_regex, str(ssoup))
-            image_url = m.group('url')
-
-            filename = image_url.split('/')[-1]
-            filename_regex = r'(?P<trunk>.+)-\d+x\d+\.(?P<ext>.+)'
-            match = re.match(filename_regex, filename)
-            if match:
-                trunk = match.group('trunk')
-                ext = match.group('ext')
-                image_url = '/'.join(
-                    image_url.split('/')[:-1]) + '/' + trunk + '.' + ext
-
-            title = ssoup.h2.text
-
-            # The element 'Telebasel Archiv' refers to the old
-            # website. We need to skip this:
-            if 'Archiv' in title:
+            try:
+                page = ssoup.attrs['href']
+                m = re.search(image_regex, str(ssoup))
+                image_url = m.group('url')
+
+                filename = image_url.split('/')[-1]
+                match = re.match(filename_regex, filename)
+                if match:
+                    trunk = match.group('trunk')
+                    ext = match.group('ext')
+                    image_url = '/'.join(
+                        image_url.split('/')[:-1]) + '/' + trunk + '.' + ext
+
+                title = ssoup.h2.text
+
+                # The element 'Telebasel Archiv' refers to the old
+                # website. We need to skip this:
+                if 'Archiv' in title:
+                    continue
+
+                urls = list(map(lambda x: x['page'], shows))
```

Comment:
If `urls` is used only to check the presence of `page` items, consider using a `set` or `frozenset` instead of a `list` because the presence check in sets is an O(1) operation vs O(n) in lists.